### ISSUE TITLE
limit minimum temperature in main branch?

### ISSUE THREAD

**Issue by elodie-kendall** [geodynamics/aspect#4958]
Hi developers,

For a while now I've been seeing temperatures going below surface T in my models, and so has Eric van der Wiel. For example I attach a prm (which is a modification to the steinberger cookbook in that I change viscosity from steinberger to visco plastic), the output file and min T which shows temperature going below 273K and even in some other models negative. After a lot of experimenting with Anne - increasing the tolerance and resolution, decreasing time step etc and still seeing these lower than surface T's we decided it was best to simply limit the minimum T. Would you want this in the main branch as it seems to be an issue in many global visco plastic models?
Elodie
[log.txt](https://github.com/geodynamics/aspect/files/9566769/log.txt)

**Comment by bangerth** [comment]
Hi @elodie-kendall:
I'm afraid nobody ever answered your question here. 

At the core of your problem are two issues: (i) For some material models, the temperature *may* actually decrease, for example due to decompression cooling (the opposite of compression heating) which you get with material models that are compressible. This shouldn't result in a drop of temperature as large as you see in the *exact* solution, but we're not computing the exact solution after all. For example, if you were on a coarse mesh, at a midocean ridge spreading center, you may get substantially negative pressures, much lower than physically reasonable. (ii) Even if you used a material description that does not allow for decreasing temperatures (say, for incompressible models without adiabatic heating, where the temperature is simply advected along and subject only to shear heating), it isn't clear that any numerical scheme also guarantees this. Indeed, in many cases one can *prove* that the solution to a PDE satisfies certain properties (say, mass conservation) but that the *numerical approximation does not* as long as you compute on a finite mesh -- with the deviation from the expected behavior larger the coarser the mesh is.

In other words, rather than paper over the issue by simply limiting the temperature, I would much rather investigate the matter a bit further.
